spec/classes/haproxy_spec.rb

```diff
@@ -463,6 +463,138 @@
         end
       end
     end
+
+    describe 'when overriding global and defaults options with user-supplied overrides and additions' do
+      # For testing the merging functionality we restrict ourselves to
+      # Debian OS family so that we don't have to juggle different sets of
+      # global_options and defaults_options (like for FreeBSD).
+      ['Debian' ].each do |osfamily|
```

Comment:
Thanks for raising that.  I'm copying the style of this kind of test above on line  373 (https://github.com/puppetlabs/puppetlabs-haproxy/pull/207/files#diff-7be58596f7d58684b9342fe66e3aeee3L373)